conbench/entities/run.py

```diff
@@ -209,6 +179,117 @@ def get_baseline_id(self):
         return run.id if run else None
 
 
+def commit_fetch_info_and_create_in_db_if_not_exists(
+    sha, repository, pr_number, branch
+) -> str:
+    """
+    Insert new Commit entity into database if required.
+
+    If Commit not yet known in database: fetch data about commit (and related
+    commits) from GitHub HTTP API if possible. Exceptions during this process
+    are logged and otherwise swallowed.
+
+    Return Commit.id (DB primary key) of existing Commit entity or of newly
+    created one. Expect database collision upon insert (in this case the ID for
+    the existing commit entity is returned).
+
+    Has slightly ~unpredictable run duration as of interaction with GitHub HTTP
+    API.
+    """
```

Comment:
That's roughly the level of docstringing we should have in the code base. Does not need to be perfect, but between fellow developers we have to communicate the intent, in particular about error handling.